components/Layout.js

```diff
@@ -2,19 +2,24 @@ import Link from 'next/link'
 import { useState, useEffect } from 'react'
 import { getUsername } from '../lib/userAuth'
 
-const Layout = ({ children }) => {
+const Layout = ({ children, id }) => {
```

Comment:
Why do we need to receive the `id` from the parent? In both of the cases where we are using Layout, id is set to `getUserId()`. I think we can just call that method inside this component.